### PR TITLE
EntitySize -> EntityDimensions

### PR DESCRIPTION
**Pull request by Sollace**
Changes for the issue at https://github.com/FabricMC/yarn/issues/715

### CODE REVIEW

**Review by Prospector — APPROVED** [review]

**Comment by liach** [comment]
please change all getter methods to getDimensions instead of getSize; same for fields/method params

**Comment by liach** [comment]
also consider renaming constant to fixed

File: mappings/net/minecraft/entity/EntityDimensions.mapping [hunk 1]
```diff
@@ -11,7 +11,7 @@ CLASS aim net/minecraft/entity/EntitySize
 	METHOD a scaled (FF)Laim;
 		ARG 1 widthRatio
 		ARG 2 heightRatio
-	METHOD b resizeable (FF)Laim;
+	METHOD b changing (FF)Laim;
 		ARG 0 x
 		ARG 1 z
 	METHOD c constant (FF)Laim;
```
Comment (ChloeDawn) [inline mappings/net/minecraft/entity/EntityDimensions.mapping]: `fixed`

**Review by ChloeDawn — CHANGES_REQUESTED** [review]

**Comment by ChloeDawn** [comment]
The parameter names for the static constructors are incorrect. They should be `width` and `height` not `x` and `z`.

File: mappings/net/minecraft/entity/EntityDimensions.mapping [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CLASS aim net/minecraft/entity/EntitySize
+CLASS aim net/minecraft/entity/EntityDimensions
 	FIELD a width F
 	FIELD b height F
 	FIELD c constant Z
```
Comment (ChloeDawn) [inline mappings/net/minecraft/entity/EntityDimensions.mapping]: `fixed`

File: mappings/net/minecraft/entity/EntityDimensions.mapping (hunk 1 above)
Comment (ChloeDawn) [reply mappings/net/minecraft/entity/EntityDimensions.mapping]: Constructor parameter and field also need to be changed to `fixed`

**Comment by ChloeDawn** [comment]
**`field`**`EnderDragonPart#partSize` -> `EnderDragonPart#dimensions`
**`field`**`Entity#size` -> `Entity#dimensions`
**`field`**`EntityType#size` -> `EntityType#dimensions`
**`field`**`EntityType$Builder#size` -> `EntityType$Builder#dimensions`
**`field`**`LivingEntity.SLEEPING_SIZE` -> `LivingEntity.SLEEPING_DIMENSIONS`
**`field`**`PlayerEntity.SIZES` -> `PlayerEntity.POSE_DIMENSIONS`
**`field`**`PlayerEntity.STANDING_SIZE` -> `PlayerEntity.STANDING_DIMENSIONS`
**`method`**`Entity#getSize(EntityPose)` -> `Entity#getDimensions(EntityPose)`
**`method`**`Entity#refreshSize` -> `Entity#refreshDimensions`
**`method`**`EntityType#getDefaultSize` -> `EntityType#getDefaultDimensions`
**`param`**`Entity#getEyeHeight@size` -> `Entity#getEyeHeight@dimensions`
**`param`**`EntityType.<init>@size` -> `EntityType.<init>@dimensions`
**`param`**`LivingEntity#getActiveEyeHeight@size` -> `LivingEntity#getActiveEyeHeight@dimensions`

**Comment by ChloeDawn** [comment]
Migration required for Fabric's `EntityTypeBuilder`:
**`field`**`size` -> `dimensions`
**`method`**`size` -> `dimensions`
**`param`**`size@size` -> `dimensions`

**Comment by kashike** [comment]
I'm not a fan of this name change - it will cause confusion with world dimensions.

**Comment by ChloeDawn** [comment]
I agree, but either way, I do not want a partial migration. All or nothing. Personally, I would call this data class `EntityShape` as it implies the width and height, and separates it away from the concept of dimensions that the game already has.

**Comment by liach** [comment]
i originally opposed this change, but i find that it will be clear if we call this one dimensions (with the s suffix) instead of dimension.

**Comment by Sollace** [comment]
@liach @kashike Again, I remind everyone that it is `Dimensions` _with an S_.

@liach I did consider doing name changes for all the getters as well, however I don't believe it's neccessary to have it use the same terminology as the class name. It's clear enough to see that `getSize()` returns a some form of Dimensions. It also avoids confusion with a method like `getDimension()`.

**Comment by Sollace** [comment]
@liach I thought people specifically didn't want that changed to fixed? (which is why it was left out of the PR)
@InsomniaKitten  (pinged wrong person)
>I believe constant is fine too, representing an immutable size. Imo fixed doesn't really imply that well.

**Comment by kashike** [comment]
@Sollace I am well aware it has an `s`, my statement is not changing.

**Comment by Sollace** [comment]
I was also thinking about world dimensions, and what's stopping us from calling them `Level`s or `LevelType`s? Minecraft has used that term in the past when referring to the overworld, so I don't see why they wouldn't still have it internally.

**Comment by liach** [comment]
how about EntityExtent?

**Comment by liach** [comment]
`EntityMeasure` would be clear and non-confusing.

See defintion 4 at https://www.dictionary.com/browse/measures (noun section)
>the extent, dimensions, quantity, etc., of something, ascertained especially by comparison with a standard: 
>to take the measure of a thing.

This covers "dimensions" "extent" in a sense while it won't be confusing as "extent" or "dimension" (both has been used in spongeapi to refer to world stuff)

@kashike @InsomniaKitten @Prospector wanna hear your opinions.

**Comment by copygirl** [comment]
> I was also thinking about world dimensions, and what's stopping us from calling them `Level`s or `LevelType`s?

The term is already standard in the entire community.
Very early on we had a discussion on this. It didn't really lead anywhere, but see #89.

----

That said, I'm split on the change.
**Pro:** `toString` clearly mentioning the actual name of the class.
**Con:** It kind of clashes with `Dimension`, for auto-complete suggestions and such.

I also don't necessarily see `Size` as a scalar value. It is often used in a 2D context to describe an element's width and height. And I don't think we should introduce another name that is different from Mojang's name.

**Comment by Sollace** [comment]
> EntityMeasure would be clear and non-confusing.

@liach Even though you didn't ask for it, I don't know if I like using `EntityMeasure`. What is it a measure of? Volume? Age? Does it specify width/height in meters, or centimeters?

Measure to me is something of standard, fixed, size or dimensions that you use to _measure_ something. Like how a meter stick is a measure of length.

**Comment by liach** [comment]
this way, dimensions is a plain scalar as well. i'd say stay with size; we should not be trolled by mojang for such a tostring statement after all.

**Comment by NeunEinser** [comment]
> this way, dimensions is a plain scalar as well.

Not sure what you are referring to. You can measure in m² (or m³), too, which to my understanding is the case when you measure dimensions since you are measuring *all* dimensions of the object.

I think Wikitonary's definition pretty much nails it and describes perfectly what this class is. (definition of a **single** dimension, not the plural form)
>A measure of spatial extent in a particular direction, such as height, width or breadth, or depth.

**Review by Runemoro — APPROVED** [review]

**Comment by Runemoro** [comment]
To me a size is the overall size of something. I wouldn't expect a class named `EntitySize` to give both a width and height.

@kashike Currently Entity.getSize (which returns its dimensions, not size!) conflicts with SlimeEntity.getSlimeSize (which correctly returns a single integer size!).

**Comment by asiekierka** [comment]
I say EntityDimensions.

**Review by LemmaEOF — APPROVED** [review]

**Review by NeunEinser — APPROVED** [review]

**Review by Gegy — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**